### ISSUE TITLE
Package installs source files to incorrect path

### ISSUE THREAD

**Issue by taiidani** [composer/composer#5561]
With the following composer.json:

```json
{
    "name": "vacasa/common",
    "autoload": {
        "files": ["autoload.php"]
    },
    "require": {
        "sabre/xml": "~1.4",
        "lstrojny/functional-php": "~1.2",
        "giggsey/libphonenumber-for-php": "~7.0",
        "authorizenet/authorizenet": "1.8.3",
        "aws/aws-sdk-php": "2.7.*",
        "geoip2/geoip2": "2.3.*",
        "nikic/fast-route": "~0.5",
        "owasp/phprbac": "2.0.*",
        "pda/pheanstalk": "3.1.*",
        "predis/predis": "~1.0",
        "twig/twig": "~1.16",
        "twig/extensions": "~1.2",
        "raven/raven": "0.10.*",
        "phpoffice/phpexcel": "1.8.1",
        "hassankhan/config": "~0.10",
        "mailgun/mailgun-php": "~1.7",
        "swiftmailer/swiftmailer": "~5.4",
        "mandrill/mandrill": "~1.0",
        "plivo/plivo-php": "^1.0",
        "illuminate/database": "5.1.*",
        "illuminate/events": "5.1.*",
        "lesstif/php-jira-rest-client": "1.7.0"
    },
    "require-dev": {
        "shanethehat/pretty-xml": "~1.0.2",
        "codeception/codeception": "~2.1",
        "jdorn/sql-formatter": "~1"
    }
}
```

Composer version: 1.1.3

When I run `composer install --no-dev`, the packages all indicate a successful installation but sometimes packages do not install correctly. Most of the time the package is [libphonenumber-for-php](https://github.com/giggsey/libphonenumber-for-php). I have [filed an issue](https://github.com/giggsey/libphonenumber-for-php/issues/124) with the maintainer and was asked to forward to the Composer team for support.

When performing a `composer install` the library appears to update, but instead of adding new code to `src/libphonenumber` it actually removes [most of] the code in `src/libphonenumber` and instead places the new code under `src/src/libphonenumber`. The presence of this redundant `src` folder in the directory tree makes the autoloader miss, and thus causes "Class 'libphonenumber\PhoneNumberUtil' not found" errors throughout our codebase after each deploy.

Completely deleting the package from `vendor` and reinstalling clean will solve the problem once, but the problem crops up again during the next `composer install`.

**Comment by alcohol** [comment]
Are you installing from dist, or source? (`--prefer-dist` vs `--prefer-source`)

By default it will try dist I believe. Can you see if using source solves the issue?
Cause that could indicate a problem with the archive that is being extracted (dist uses archives).

**Comment by taiidani** [comment]
I'm pretty sure the process uses dist -- we don't specify either parameter and the logs are clean.

```
Loading composer repositories with package information
Installing dependencies from lock file
Removing giggsey/libphonenumber-for-php (7.4.4)
Installing giggsey/libphonenumber-for-php (7.4.5)
Downloading: Connecting...
Downloading: 0% (etc etc) 100%
```

We can try again with `--prefer-source` although with the problem cropping up only occasionally it may take a while to reproduce. We'll also turn on more verbose logging to see if we can catch more helpful information.

**Comment by taiidani** [comment]
Sorry for the long pause on this one but I think we figured it out. It turns out that the failure was due to some rubberbanding between our two web servers. We push our files to Web1 and run Composer on it while letting the files sync across to Web2.

The syncing tool used is https://github.com/axkibe/lsyncd. Something with our two-way sync propagated changes back to Web1 and caused the corrupted data.

tl;dr, not any problem with Composer. I hope this helps anyone else that runs into the behavior!
